miner/algo_greedy.go

```diff
@@ -20,17 +20,22 @@ type greedyBuilder struct {
 	chainData        chainData
 	builderKey       *ecdsa.PrivateKey
 	interrupt        *int32
+	algoConf         algorithmConfig
 }
 
 func newGreedyBuilder(
-	chain *core.BlockChain, chainConfig *params.ChainConfig,
+	chain *core.BlockChain, chainConfig *params.ChainConfig, algoConf *algorithmConfig,
 	blacklist map[common.Address]struct{}, env *environment, key *ecdsa.PrivateKey, interrupt *int32,
 ) *greedyBuilder {
+	if algoConf == nil {
```

Comment:
I'd suggest requiring algoConf to be non-nil, and instead passing `&defaultAlgorithmConfig` wherever default should be used

Reply:
didn't address in this PR but I made that change in another PR because it was relevant for some other changes